# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/x11/xfree86/startwm.patch
# ROCK Linux is Copyright (C) 1998 - 2003 Clifford Wolf
# 
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version. A copy of the GNU General Public
# License can be found at Documentation/COPYING.
# 
# Many people helped and are helping developing ROCK Linux. Please
# have a look at http://www.rocklinux.org/ and the Documentation/TEAM
# file for details.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- ./programs/twm/system.twmrc.orig	Fri Jan 21 19:41:50 2000
+++ ./programs/twm/system.twmrc	Thu Dec 28 15:23:15 2000
@@ -12,6 +12,8 @@
 # $XFree86: xc/programs/twm/system.twmrc,v 1.3 2000/01/21 18:41:50 dawes Exp $
 #
 
+NoDefaults
+
 NoGrabServer
 RestartPreviousState
 DecorateTransients
@@ -21,23 +23,26 @@
 IconFont "-adobe-helvetica-bold-r-normal--*-100-*-*-*-*-*-*"
 IconManagerFont "-adobe-helvetica-bold-r-normal--*-100-*-*-*"
 #ClientBorderWidth
+RandomPlacement
+AutoRelativeResize
+IconifyByUnmapping
 
 Color
 {
-    BorderColor "slategrey"
-    DefaultBackground "rgb:2/a/9"
-    DefaultForeground "gray85"
-    TitleBackground "rgb:2/a/9"
-    TitleForeground "gray85"
-    MenuBackground "rgb:2/a/9"
-    MenuForeground "gray85"
-    MenuTitleBackground "gray70"
-    MenuTitleForeground "rgb:2/a/9"
-    IconBackground "rgb:2/a/9"
-    IconForeground "gray85"
-    IconBorderColor "gray85"
-    IconManagerBackground "rgb:2/a/9"
-    IconManagerForeground "gray85"
+	 BorderColor "slategrey"
+	 DefaultBackground "rgb:2/a/9"
+	 DefaultForeground "gray85"
+	 TitleBackground "rgb:2/a/9"
+	 TitleForeground "gray85"
+	 MenuBackground "rgb:2/a/9"
+	 MenuForeground "gray85"
+	 MenuTitleBackground "gray70"
+	 MenuTitleForeground "rgb:2/a/9"
+	 IconBackground "rgb:2/a/9"
+	 IconForeground "gray85"
+	 IconBorderColor "gray85"
+	 IconManagerBackground "rgb:2/a/9"
+	 IconManagerForeground "gray85"
 }
 
 #
@@ -46,48 +51,65 @@
 MoveDelta 3
 Function "move-or-lower" { f.move f.deltastop f.lower }
 Function "move-or-raise" { f.move f.deltastop f.raise }
-Function "move-or-iconify" { f.move f.deltastop f.iconify }
+Function "xiconify" { f.iconify f.showiconmgr }
 
 #
 # Set some useful bindings.  Sort of uwm-ish, sort of simple-button-ish
 #
-Button1 = : root : f.menu "defops"
 
-Button1 = m : window|icon : f.function "move-or-lower"
-Button2 = m : window|icon : f.iconify
-Button3 = m : window|icon : f.function "move-or-raise"
+LeftTitleButton "opendotMask" = f.function "xiconify"
+RightTitleButton "star" = f.delete
+
+Button1 = : root : f.menu "defops"
+Button2 = : root : f.menu "apps"
+Button3 = : root : f.exec "xterm &"
 
 Button1 = : title : f.function "move-or-raise"
-Button2 = : title : f.raiselower
+Button2 = : title : f.fullzoom
+Button3 = : title : f.function "move-or-lower"
 
-Button1 = : icon : f.function "move-or-iconify"
-Button2 = : icon : f.iconify
+Button1 = : icon : f.function "xiconify"
+Button3 = : icon : f.identify
+
+Button1 = : iconmgr : f.function "xiconify"
+Button2 = : iconmgr : f.raise
+Button3 = : iconmgr : f.identify
+
+Button1 = : frame : f.resize
+Button2 = : frame : f.raiselower
+Button3 = : frame : f.function "move-or-raise"
 
-Button1 = : iconmgr : f.iconify
-Button2 = : iconmgr : f.iconify
 
 #
 # And a menus with the usual things
 #
 menu "defops"
 {
-"Twm"	f.title
-"Iconify"	f.iconify
-"Resize"	f.resize
-"Move"		f.move
-"Raise"		f.raise
-"Lower"		f.lower
-""		f.nop
-"Focus"		f.focus
-"Unfocus"	f.unfocus
-"Show Iconmgr"	f.showiconmgr
-"Hide Iconmgr"	f.hideiconmgr
-""		f.nop
-"Xterm"		f.exec "exec xterm &"
-""		f.nop
-"Kill"		f.destroy
-"Delete"	f.delete
-""		f.nop
-"Restart"	f.restart
-"Exit"		f.quit
+	"TWM"		f.title
+	"Iconify"	f.iconify
+	"Resize"	f.resize
+	"Move"		f.move
+	"Raise"		f.raise
+	"Lower"		f.lower
+	"Zoom"		f.fullzoom
+	""		f.nop
+	"Focus"		f.focus
+	"Unfocus"	f.unfocus
+	"Show Iconmgr"	f.showiconmgr
+	"Hide Iconmgr"	f.hideiconmgr
+	""		f.nop
+	"Kill"		f.destroy
+	"Delete"	f.delete
+	""		f.nop
+	"Restart"	f.restart
+	"Exit"		f.quit
+}
+
+menu "apps"
+{
+	"Applications"		f.title
+	"xTerm"			f.exec "xterm &"
+	"xTerm (black)"		f.exec "xterm -fg white -bg black &"
+	"Mozilla"		f.exec "mozilla &"
+	"Icon Manager"		f.showiconmgr
 }
--- ./programs/xdm/config/Xsession.orig	Sun Jan 11 04:48:32 1998
+++ ./programs/xdm/config/Xsession	Thu Dec 28 15:23:15 2000
@@ -33,5 +33,11 @@
 	if [ -r "$resources" ]; then
 		xrdb -load "$resources"
 	fi
+
+	WINDOWMANAGER="${WINDOWMANAGER:-startkde}"
+	[ "$WINDOWMANAGER" =  "gnome" ] && WINDOWMANAGER="gnome-session"
+	[ "$WINDOWMANAGER" =  "kde"   ] && WINDOWMANAGER="startkde"
+	[ "$WINDOWMANAGER" != "twm"   ] && exec $WINDOWMANAGER
+	xsetroot -solid darkcyan ; exec twm ; exec xterm
 	exec xsm
 fi
--- ./programs/xdm/config/Xsetup_0.orig	Wed Apr 27 09:20:03 1994
+++ ./programs/xdm/config/Xsetup_0	Thu Dec 28 15:23:15 2000
@@ -1,3 +1,4 @@
 #!/bin/sh
 # $XConsortium: Xsetup_0,v 1.3 93/09/28 14:30:31 gildea Exp $
 xconsole -geometry 480x130-0-0 -daemon -notify -verbose -fn fixed -exitOnFail
+xsetroot -solid black
--- ./programs/xinit/xinitrc.cpp.orig	Wed Apr 27 09:13:07 1994
+++ ./programs/xinit/xinitrc.cpp	Thu Dec 28 15:23:15 2000
@@ -24,10 +24,10 @@
     xmodmap $usermodmap
 fi
 
-XCOMM start some nice programs
+XCOMM start the window manager
 
-twm &
-xclock -geometry 50x50-1+1 &
-xterm -geometry 80x50+494+51 &
-xterm -geometry 80x20+494-0 &
-exec xterm -geometry 80x66+0+0 -name login
+WINDOWMANAGER="${WINDOWMANAGER:-startkde}"
+[ "$WINDOWMANAGER" =  "gnome" ] && WINDOWMANAGER="gnome-session"
+[ "$WINDOWMANAGER" =  "kde"   ] && WINDOWMANAGER="startkde"
+[ "$WINDOWMANAGER" != "twm"   ] && exec $WINDOWMANAGER
+xsetroot -solid darkcyan ; exec twm ; exec xterm
